migration.go

```
@@ -202,52 +207,63 @@ func (m *Morpher) Run(db *sql.DB) error {
 			continue
 		}
 
 		m.Log.Info("applying migration", slog.String("file", migration.Key()))
 		startMigration = time.Now()
+
+		// Check context before starting a transaction
+		if err := ctx.Err(); err != nil {
+			return fmt.Errorf("context cancelled before migration %s: %w", migration.Key(), err)
+		}
+
 		tx, txBeginErr := db.Begin()
```

golangci-lint (2.2.2)

(*database/sql.DB).Begin must not be called. use (*database/sql.DB).BeginTx

(noctx)